From 8c489a3254cc1fc5bb116bfb266ec0ed06879d70 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 7 May 2018 20:08:45 +0200
Subject: [PATCH 1/1] tools/file2include: create Linux style SPDX header

file2include is used to convert a binary file to a C include.
With the patch the SPDX header is written to the first line as
expected by scripts/checkpatch.pl.

Cf. https://www.kernel.org/doc/html/v4.16/process/license-rules.html

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/file2include.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/file2include.c b/tools/file2include.c
index 7ca45c8339a..e602f937f15 100644
--- a/tools/file2include.c
+++ b/tools/file2include.c
@@ -62,12 +62,11 @@ int main(int argc, char *argv[])
 	count = fread(buf, 1, count, file);
 
 	/* Generate output */
+	printf("/* SPDX-License-Identifier: GPL-2.0+ */\n");
 	printf("/*\n");
 	printf(" *  Non-zero %u byte strings of a disk image\n", BLOCK_SIZE);
 	printf(" *\n");
 	printf(" *  Generated with tools/file2include\n");
-	printf(" *\n");
-	printf(" *  SPDX-License-Identifier:	GPL-2.0+\n");
 	printf(" */\n\n");
 	printf("#define EFI_ST_DISK_IMG { 0x%08zx, { \\\n", count);
 
-- 
2.17.0

